Source/DafnyCore/Rewriters/ExpectContracts.cs

```diff
@@ -204,9 +204,17 @@ public class ExpectContracts : IRewriter {
     var post = new AttributedExpression(new BinaryExpr(tok, BinaryExpr.Opcode.Eq, r, fn) {
       Type = Type.Bool
     });
+    // If f.Reads is empty, replace it with an explicit `reads {}` so that we don't replace that
```

Comment:
I think this code becomes obsolete by assigning the default `reads *` during parsing